nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -37,6 +37,8 @@ public final class NessieUtil {
   public static final String NESSIE_CONFIG_PREFIX = "nessie.";
   static final String APPLICATION_TYPE = "application-type";
 
+  public static final String CLIENT_API_VERSION = "client-api-version";
```

Comment:
better to define in `NessieConfigConstants` on Nessie side. But that needs a Nessie release.

Reply:
`nessie.client-builder-impl` is also a client's configuration. But we declare that constant variable in Nessie repo and use it here. So, I thought we can have it similarly.

Reply:
Actually, it might be possible to have a constant for the API version on the Nessie Client side, but then the client should be choosing the class object for the `build(...)` call. This might be an option for a smooth migration path from API v1 to v2... but certainly it requires some Nessie-side changes.

Reply:
I think it is fine to keep this constant in this class for now.